geoarrow-pandas/src/geoarrow/pandas/lib.py

```diff
@@ -191,7 +191,16 @@ def isna(self):
             return out.to_numpy(zero_copy_only=False)
 
     def copy(self):
-        return GeoArrowExtensionArray._from_sequence(self, dtype=self._dtype)
+        if isinstance(self._data, _pa.ChunkedArray):
```

Comment:
Too late, but I think it would be good to add a comment about why you are concatting the underlying array for the copy() method, that's not necessarily obvious for the reader I think